Cython/Compiler/FusedNode.py

```diff
@@ -775,45 +767,13 @@ def __pyx_fused_cpdef(signatures, args, kwargs, defaults, _fused_sigindex_ref=[N
 
         pyx_code.put_chunk(
             """
-                sigindex_matches = []
-                sigindex_candidates = [fused_sigindex]
-
-                for dst_type in dest_sig:
-                    found_matches = []
-                    found_candidates = []
-                    # Make two separate lists: One for signature sub-trees
-                    #        with at least one definite match, and another for
-                    #        signature sub-trees with only ambiguous matches
-                    #        (where `dest_sig[i] is None`).
-                    if dst_type is None:
-                        for sn in sigindex_matches:
-                            found_matches.extend((<dict> sn).values())
-                        for sn in sigindex_candidates:
-                            found_candidates.extend((<dict> sn).values())
-                    else:
-                        for search_list in (sigindex_matches, sigindex_candidates):
-                            for sn in search_list:
-                                type_match = (<dict> sn).get(dst_type)
-                                if type_match is not None:
-                                    found_matches.append(type_match)
-                    sigindex_matches = found_matches
-                    sigindex_candidates = found_candidates
-                    if not (found_matches or found_candidates):
-                        break
-
-                candidates = sigindex_matches
-
-                if not candidates:
-                    raise TypeError("No matching signature found")
-                elif len(candidates) > 1:
-                    raise TypeError("Function call with ambiguous argument types")
-                else:
-                    return (<dict>signatures)[candidates[0]]
-            """)
+                return __pyx_ff_match_signatures(<dict> signatures, dest_sig, fused_sigindex)
```

Comment:
I think `fused_sigindex` is a `dict` here that you're passing to something expecting a list.